app/models/manageiq/providers/ansible_tower/shared/automation_manager/configuration_script_source.rb

```diff
@@ -14,4 +14,18 @@ def provider_collection(manager)
   def provider_object(connection = nil)
     (connection || connection_source.connect).api.projects.find(manager_ref)
   end
+
+  REFRESH_ON_TOWER_SLEEP = 1.second
+  def refresh_in_provider
+    with_provider_object do |project|
+      return unless project.can_update?
+
+      project_update = project.update
+
+      while project_update.finished.blank?
```

Comment:
The `refresh_in_tower` only updates _one_ project. But you are right, this should timeout...
 
https://github.com/ManageIQ/manageiq/pull/14405 would be a great solution to this. But thats not merged and we need a quick way to solve https://bugzilla.redhat.com/show_bug.cgi?id=1441232